modules/k6/examples_test.go

```diff
@@ -0,0 +1,84 @@
+package k6_test
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/k6"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func ExampleRunContainer() {
+	// runK6Container {
+	ctx := context.Background()
+
+	// create a container with the httpbin application that will be the target
+	// for the test script that runs in the k6 container
+	gcr := testcontainers.GenericContainerRequest{
+		ProviderType: testcontainers.ProviderDocker,
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image: "kennethreitz/httpbin",
+			ExposedPorts: []string{
+				"80",
+			},
+			WaitingFor: wait.ForExposedPort(),
+		},
+		Started: true,
+	}
+	httpbin, err := testcontainers.GenericContainer(ctx, gcr)
+	if err != nil {
+		panic(fmt.Errorf("failed to create httpbin container %w", err))
+	}
+
+	defer func() {
+		if err := httpbin.Terminate(ctx); err != nil {
+			panic(fmt.Errorf("failed to terminate container: %w", err))
+		}
+	}()
+
+	httpbinIP, err := httpbin.ContainerIP(ctx)
+	if err != nil {
+		panic(fmt.Errorf("failed to get httpbin IP: %w", err))
+	}
+
+	absPath, err := filepath.Abs(filepath.Join("scripts", "httpbin.js"))
+	if err != nil {
+		panic(fmt.Errorf("failed to get path to test script: %w", err))
+	}
+
+	// run the httpbin.js test scripts passing the IP address the httpbin container
+	k6, err := k6.RunContainer(
+		ctx,
+		k6.WithTestScript(absPath),
+		k6.SetEnvVar("HTTPBIN", httpbinIP),
+	)
+	if err != nil {
+		panic(fmt.Errorf("failed to start k6 container: %w", err))
+	}
+
+	defer func() {
+		if err := k6.Terminate(ctx); err != nil {
+			panic(fmt.Errorf("failed to terminate container: %w", err))
+		}
+	}()
+
+	// assert the result of the test
+	state, err := k6.State(ctx)
+	if err != nil {
+		panic(err)
+	}
+	if state.ExitCode != 0 {
+		logs, err := k6.Logs(ctx)
+		if err != nil {
+			panic(err)
+		}
+
+		fmt.Println("k6 test failed:")
```

Comment:
IIRC, testable examples need fmt prints and comments with the output right?

I was more on fmt-printing the exit code instead of the panic.

Reply:
I'm not familiar with testable examples and maybe I misunderstood your previous comment.

Could you please indicate what is exactly required in this case? All examples from other modules simply panic on any error.

Reply:
Please look at https://github.com/testcontainers/testcontainers-go/blob/3079ce0890d15d82d45540ae476aec2b040d64e6/modules/clickhouse/examples_test.go#L46-L62

- L46 prints if the state of the container is Running, matched in L61
- L58 prints if the connection string includes "clickhouse-go" in L62

We are panicking every time we are not performing an assertion. For the assertions, we just print a value and add a comment under `// Output:`. Please see https://go.dev/blog/examples

So for this case, I'd print out the container state with fmt, and in the comment the expected state code. Or if the container log is deterministic, then print out the logs (very similar to what you already did)